models/Comment.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Comment Model
+ *
+ * Stores user comments on projects, with support-ish for threaded replies.
+ * Uses custom `isDeleted` flag so that deleted comments/users can still show up (e.g. "[deleted]")
+ * and keep their reply structure intact.
+ */
+
+module.exports = (sequelize, Sequelize) => {
+  const { DataTypes } = Sequelize;
+
+  const Comment = sequelize.define(
+    "Comment",
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        autoIncrement: true,
+        primaryKey: true,
+      },
+      userId: {
+        type: DataTypes.UUID,
+        allowNull: true, // Allows for soft deletion
+      },
+      projectId: {
+        type: DataTypes.UUID,
+        allowNull: false,
+      },
+      message: {
+        type: DataTypes.TEXT, // NOTE: Diagram says STRING. Use TEXT for lengthy comments
+        allowNull: false,
+      },
```

Comment:
We could potentially set something like this up here

```suggestion
      message: {
        type: DataTypes.TEXT, // NOTE: Diagram says STRING. Use TEXT for lengthy comments
        allowNull: false,
        get() {
          const rawValue = this.getDataValue("message");
          return this.getDataValue("isDeleted") ? "[deleted]" : rawValue;
        },
      },
```

Reply:
We would need to handle both deleted comment and deleted user. So it could be:
```suggestion
   get() {
          const isCommentDeleted = this.getDataValue("isDeleted");
          const userId = this.getDataValue("userId");
          const rawValue = this.getDataValue("message");
          return isCommentDeleted || userId === null ? "[deleted]" : rawValue;
        },

```
But we should probably separate the logic, and have the formatting in its own file (e.g. utilities), or front-end could handle it. So, either have both deleted user and deleted comment/message elsewhere, or keep your approach here and handle deleted user(if(comment.userId === null)) elsewhere.

Reply:
I think this might be a point for personal opinion and we can put it to a bit of a vote perhaps.  @johnDavid97 @DavidHansen84 what are your thoughts on this?

---

### Option 1:
Shows [deleted] if the comment is deleted OR the user is deleted.
Hides the original message in both cases.

---

### Option 2:
Only shows [deleted] when the comment itself is flagged as deleted.
Keeps the message visible even if the user account is gone.

--- 

I think option 2, preserve the conversation to keep things coherent unless a user explicitly wants to delete their comments before they delete themselves. Most platforms I have seen this approach used when a user is shown as deleted but their comment is preserved.

Reply:
I would go for option 2 :)

Reply:
Since preserving comments are ok, I agree on option 2. Do we want to handle the transformation elsewhere then?

Reply:
Yes option 2